contracts/ToucanRegenBridge.sol

```diff
@@ -63,11 +63,25 @@ contract ToucanRegenBridge is Ownable, Pausable, AccessControl {
     // ----------------------------------------
 
     modifier isRegenAddress(bytes calldata account) {
-        // verification: checking if account starts with "regen1"
-        require(account.length >= 44, "regen address is at least 44 characters long");
+        // verification: address length is 44 (standard) or 64 (derived)
+        bool length = account.length == 44 || account.length == 64;
```

Comment:
Can you cache `account.length` to `accountLen` and use the latter throughout the code instead of the former? 
```solidity
uint256 accountLen = account.length;
```
Just a small gas optimization. 

I'd also say that then it's better to get rid of `length` and inline the OR check in the `require` statement which is more common.
```solidity
require(accountLen == 44 || accountLen == 64, "regen address must be 44 or 64 chars");
```

Reply:
Thanks! Updated https://github.com/regen-network/toucan-bridge/pull/49/commits/a26f1dd339ecec76c186cc4bdcd563441bb09a0b.